engine/src/main/java/org/terasology/math/VectorPools.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.math;
+
+import org.terasology.math.geom.Vector3i;
+
+public class VectorPools {
+
+    private static final int POOL_SIZE = 100;
+
+    private static int currentVector3isAvailable;
+    private static final Vector3i[] VECTOR_3I_POOL = new Vector3i[POOL_SIZE];
```

Comment:
I think this should be `Vector3iPool` and not `VECTOR_3I_POOL`.

(Note that this variable is a constant reference, and not a constant itself. A small - but important - distinction)